### PR TITLE
feat(objects): implement objects module

### PR DESCRIPTION
**Pull request by tourlbr**
Note: A new module named "ObjectsModule" was made. However the linter was complaining about using "Object" as a type so I had to use "ObjectMedia".

Screenshot 2023-01-25 at 14 20 59

### CODE REVIEW

File: src/modules/objects/objects.types.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+	alternateName: string;
+	description: string;
+	schemaIdentifier: string;
+	dctermsFormat: string;
+	transcript: string;
+	dateCreated: string;
+	files: ObjectFile[];
+}
+
+export interface ObjectMedia {
```
Comment (bertyhell) [inline src/modules/objects/objects.types.ts:49]: Let's use IeObject if Object throws an error
then it is closer to the database naming: objects.ie
ie => Intellectual Entity

File: src/modules/objects/services/objects.service.ts [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+				hit._source.schema_thumbnail_url =
+					await this.playerTicketService.resolveThumbnailUrl(
+						hit._source.schema_thumbnail_url,
+						referer
+					);
+				return hit;
+			})
+		);
+
+		return esResponse;
```
Comment (bertyhell) [inline src/modules/objects/services/objects.service.ts:87]: the other fields can also be simplified, so we don't expose elasticsearch format to the client:
See task:
```
  "aggregations": {...},
  "items": IeObject[],
  "total": number,
```

File: src/modules/objects/controllers/objects.controller.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import { ObjectsService } from '../services/objects.service';
+
+import { TranslationsService } from '~modules/translations/services/translations.service';
+import { SessionUserEntity } from '~modules/users/classes/session-user';
+import { Permission } from '~modules/users/types';
+import { RequireAllPermissions } from '~shared/decorators/require-permissions.decorator';
+import { SessionUser } from '~shared/decorators/user.decorator';
+
+@ApiTags('Objects')
+@Controller('objects')
```
Comment (bertyhell) [inline src/modules/objects/controllers/objects.controller.ts:18]: maybe also change this then to:
```suggestion
@Controller('ie-objects')
```

File: src/modules/objects/controllers/objects.controller.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+export class ObjectsController {
```
Comment (bertyhell) [inline src/modules/objects/controllers/objects.controller.ts:19]:
```suggestion
export class IeObjectsController {
```

**Review by bertyhell — APPROVED** [review]

File: src/modules/objects/objects.types.ts (hunk 1 above)
Comment (tourlbr) [reply src/modules/objects/objects.types.ts]: Adjusted

File: src/modules/objects/controllers/objects.controller.ts (hunk 3 above)
Comment (tourlbr) [reply src/modules/objects/controllers/objects.controller.ts]: Adjusted

File: src/modules/objects/controllers/objects.controller.ts (hunk 4 above)
Comment (tourlbr) [reply src/modules/objects/controllers/objects.controller.ts]: Adjusted

**Pull request closed without merging**